runelite-client/src/main/java/net/runelite/client/plugins/timersandbuffs/BuffCounter.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2024, YvesW <https://github.com/YvesW>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.timersandbuffs;
+
+import java.awt.Color;
+import lombok.AccessLevel;
+import lombok.Getter;
+import net.runelite.client.ui.overlay.infobox.Counter;
+
+@Getter(AccessLevel.PACKAGE)
+class BuffCounter extends Counter
+{
+	private final TimersAndBuffsPlugin plugin;
+	private final GameCounter gameCounter;
+
+	BuffCounter(
+		TimersAndBuffsPlugin plugin,
+		GameCounter gameCounter,
+		int count)
+	{
+		super(null, plugin, count);
+		this.plugin = plugin;
+		this.gameCounter = gameCounter;
+	}
+
+	@Override
+	public Color getTextColor()
+	{
+		switch (gameCounter.getColorBoundaryType())
+		{
+			case LESS_THAN_EQUAL_TO:
+				return getCount() <= gameCounter.getBoundary() ? gameCounter.getColor() : Color.WHITE;
```

Comment:
These could be made into Predicate methods of the BoundaryType enum fields to simplify this getTextColor implementation

Reply:
Apologies, I don't have a lot of experience with using Predicates. I tried using Predicate [here](https://github.com/YvesW/runelite/commit/2c57d1cac0ba6bd1d3fd41d7af442ff07ec25d86) but failed. In the end I used BiPredicate, but I doubt that is what you meant.

Reply:
No, that looks correct--I hadn't thought about it needing two args instead of one.